### PR TITLE
Move slow lint checks to CI-only

### PR DESCRIPTION
**Pull request by amacneil**
before

```
Rule                           | Time (ms) | Relative
:------------------------------|----------:|--------:
import/no-self-import          | 24012.591 |    15.4%
prettier/prettier              | 23319.138 |    15.0%
react/no-direct-mutation-state | 15868.722 |    10.2%
react/prop-types               | 14142.034 |     9.1%
react/no-deprecated            | 13845.635 |     8.9%
react/display-name             | 13580.431 |     8.7%
react/no-string-refs           | 13509.158 |     8.7%
react/require-render-return    | 13492.359 |     8.7%
react/no-unused-prop-types     | 13315.296 |     8.6%
✔ Lint done...
yarn lint  250.16s user 14.20s system 130% cpu 3:23.06 total
```

after

```
Rule                            | Time (ms) | Relative
:-------------------------------|----------:|--------:
import/order                    | 16906.047 |    15.2%
react/no-direct-mutation-state  | 13731.133 |    12.4%
react/prop-types                | 11963.317 |    10.8%
react/no-deprecated             | 11807.723 |    10.6%
react/display-name              | 11604.820 |    10.4%
react/no-string-refs            | 11533.362 |    10.4%
react/no-unused-prop-types      | 11512.345 |    10.4%
react/require-render-return     | 11503.888 |    10.4%
import/no-useless-path-segments |  2209.603 |     2.0%
✔ Lint done...
yarn lint  168.43s user 11.71s system 118% cpu 2:32.32 total
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
           # Specify names so that the GitHub branch protection settings for
           # required checks don't need to change if we ever change the commands used.
           - name: lint
-            command: yarn run lint:nofix
```
Comment (defunctzombie) [inline .github/workflows/ci.yml:21]: `lint:full` ?

**Comment by defunctzombie** [comment]
You can likely also drop:

react/no-unused-prop-types -> we use typescript, not prop-types
react/prop-types -> we use typescript, not prop-types

react/require-render-return -> I think typescript will catch these errors

**Comment by defunctzombie** [comment]
This is awesome - it will make local dev better by not performing the same operation multiple times when we "lint on save" in vscode.

**Comment by defunctzombie** [comment]
`react/no-unused-prop-types` is useful - it is catching unused properties since typescript is not catching these errors

**Comment by amacneil** [comment]
Disabled `react/prop-types` entirely since it is a useless check, it was only flagging two files and in both cases typescript was capable of inferring the correct type.

I am going to fix `react/no-unused-prop-types` in a separate PR and turn it into an error.

**Pull request merged**